cdk_framework/EKS/lib/utils/validate-cluster-config.ts

```diff
@@ -0,0 +1,146 @@
+const supportedFields = new Set(['version', 'ec2_instance', 'launch_type', 'node_size'])
+const supportedVersions = new Set(['1.18', '1.19', '1.20', '1.21']);
+const supportedCPUArchitectures = new Set(['m5', 'm6g', 't4g']);
+const supportedNodeSizes = new Set(['medium', 'large', 'xlarge', '2xlarge', '4xlarge', '8xlarge', '12xlarge', '16xlarge', '24xlarge', 'metal']);
+const supportedT4gInstances = new Set(['nano', 'micro', 'small', 'medium', 'large', 'xlarge', '2xlarge'])
+const requiredFields = new Set(['version', 'launch_type'])
+
+
+export function validateClustersConfig(info: unknown){
+    const data = Object(info)
+    if(!data['clusters']){
+        throw new Error('No clusters field being filed in the yaml file')
+    }
+    const clusterInfo = data['clusters']
+    const clusterNamesSet = new Set()
+    for(const [key, value] of Object.entries(clusterInfo)){
+        if(Object.keys(Object(value)).length !== 2){
+            throw new Error('Did not use proper fields for cluster. You can only have launch_type and version')
+        }
+        const val = Object(value)
+        if(clusterNamesSet.has(key)){
+            throw new Error('Cannot have multiple clusters with the samne name')
+        }
+        clusterNamesSet.add(key)
+        validateRequiredFields(val)
+        for(const [k, v] of Object.entries(val)){
+            if(!supportedFields.has(k)){
+                throw new Error('Incompatible field type')
+            }
+            switch(k){
+                case 'version':
+                    val[k] = validateVersion(String(v))
+                    break;
+                case 'launch_type':
+                    val[k] = convertAndValidateLaunchType(val)
+                    break
+            }
+        }
+    }
+}
+
+function validateRequiredFields(fields: any){
+    for(const field of requiredFields){
+        if(!fields[field]){
+            throw new Error('Required field - ' + field + ' - not provided')
+        }
+    }
```

Comment:
```suggestion
    for(const expectedField of requiredFields){
        if(!fields[expectedField]){
            throw new Error('Required field - ' + expectedField + ' - not provided')
        }
    }
```
nit: This may improve readability